src/main/java/sg/ncl/MainController.java

```diff
@@ -3494,6 +3486,514 @@ public String adminEnergy(Model model,
+                switch (exceptionState) {
+                    case PROJECT_DETAILS_NOT_FOUND_EXCEPTION:
+                        log.warn("Project not found for deleting");
+                        attr.addFlashAttribute(MESSAGE, "Error(s):<ul><li>project not found for deleting</li></ul>");
+                        break;
+                    case FORBIDDEN_EXCEPTION:
+                        log.warn("Removing of project forbidden.");
+                        attr.addFlashAttribute(MESSAGE, "Error(s):<ul><li>deleting project forbidden</li></ul>");
+                        break;
+                    default:
+                        log.warn("Unknown error for validating project.");
+                        attr.addFlashAttribute(MESSAGE, "Error(s):<ul><li>unknown error for deleting project</li></ul>");
+                }
+            } else {
+                log.info("Project details deleted: {}", responseBody);
+            }
+        } catch (IOException e) {
+            log.error("adminMonthlyRemove: {}", e.toString());
+            throw new WebServiceRuntimeException(e.getMessage());
+        }
+
+        return "redirect:/admin/monthly";
+    }
+
+    @GetMapping("/admin/monthly/{id}/usage")
+    public String adminMonthlyUsage(@PathVariable String id, HttpSession session, Model model) {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        ResponseEntity response = restTemplate.exchange(properties.getMonthly() + "/" + id, HttpMethod.GET, request, String.class);
+        JSONObject jsonObject = new JSONObject(response.getBody().toString());
+        ProjectDetails projectDetails = getProjectDetails(jsonObject);
+        model.addAttribute("project", projectDetails);
+
+        return "admin_monthly_usage";
+    }
+
+    @GetMapping(value = {"/admin/monthly/{id}/usage/contribute", "/admin/monthly/{id}/usage/contribute/{month}"})
+    public String adminMonthlyUsageContribute(@PathVariable String id, @PathVariable Optional<String> month,
+                                              HttpSession session, Model model) {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        if (month.isPresent()) {
+            HttpEntity<String> request = createHttpEntityHeaderOnly();
+            ResponseEntity response = restTemplate.exchange(properties.getMonthlyUsage(id) + "/" + month.get(), HttpMethod.GET, request, String.class);
+            JSONObject usage = new JSONObject(response.getBody().toString());
+            JSONObject usageId = usage.getJSONObject("id");
+            ProjectUsage projectUsage = new ProjectUsage();
+            projectUsage.setId(usageId.getInt("projectDetailsId"));
+            projectUsage.setMonth(usageId.getString("monthYear"));
+            projectUsage.setUsage(usage.getInt("monthlyUsage"));
+            model.addAttribute("usage", projectUsage);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "usage" 6 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)